build/builder.go

```diff
@@ -505,15 +505,44 @@ func (o *Options) FindAllShards() []string {
 
 // IgnoreSizeMax determines whether the max size should be ignored.
 func (o *Options) IgnoreSizeMax(name string) bool {
+	matched := false
 	for _, pattern := range o.LargeFiles {
 		pattern = strings.TrimSpace(pattern)
-		m, _ := doublestar.PathMatch(pattern, name)
+		negated, validatedPattern := checkIsNegatePattern(pattern)
+
+		m, _ := doublestar.PathMatch(validatedPattern, name)
 		if m {
-			return true
+			// match against a negated LargeFiles pattern overrides all non-exclude matches
+			if negated {
+				return false
+			}
+			matched = true
 		}
 	}
 
-	return false
+	return matched
+}
+
+func checkIsNegatePattern(pattern string) (bool, string) {
+	negate := "!"
+
+	if len(pattern) < len(negate) {
+		return false, pattern
+	}
+
+	// if negated then strip prefix meta characters representing negated filter pattern
+	if strings.Index(pattern, negate) == 0 {
+		return true, pattern[len(negate):]
+	}
+
+	escape := "\\"
```

Comment:
Do we need the double slash? Can we just go with "\\"?

Reply:
I need the string literal of `\` since we are handling our own escape char interpretation for escaping the negate string (in this case `!`), so I need to escape the backslash with a backslash, or I could use back tick like:

`escape := `\``

back tick doesn't seem to be used as often as `" "` or `' '`.

We also have to double slash in site admin as well, although that's to appease the site admin parser not my golang interpreter.